src/components/builder/values/values-config.stories.tsx

```diff
@@ -0,0 +1,201 @@
+import {Form, Formik} from 'formik';
+
+import {withFormik} from '@/sb-decorators';
+
+import ValuesConfig from './values-config';
+
+export default {
+  title: 'Formio/Builder/Values/ValuesConfig',
+  component: ValuesConfig,
+  parameters: {
+    controls: {hideNoControlsWarning: true},
+    modal: {noModal: true},
+    formik: {
+      initialValues: {
+        openForms: {
+          dataSrc: '',
+        },
+      },
+    },
+  },
+  args: {
+    name: 'values',
+  },
+  argTypes: {
+    name: {control: {disable: true}},
+  },
+  tags: ['autodocs'],
+} as Meta<typeof ValuesConfig>;
+
+type SelectboxesStory = StoryObj<typeof ValuesConfig<SelectboxesComponentSchema>>;
+type RadioStory = StoryObj<typeof ValuesConfig<RadioComponentSchema>>;
+
+/**
+ * Variant pinned to the `SelectboxesComponentSchema` component type.
+ */
+export const SelectBoxes: SelectboxesStory = {
+  decorators: [withFormik],
+};
+
+export const SelectBoxesManual: SelectboxesStory = {
+  decorators: [withFormik],
+  parameters: {
+    formik: {
+      initialValues: {
+        openForms: {
+          dataSrc: 'manual',
+        },
+        values: [
+          {
+            value: 'a',
+            label: 'A',
+          },
+          {
+            value: 'b',
+            label: 'B',
+          },
+        ],
+      },
+    },
+  },
+};
+
+export const SelectBoxesVariable: SelectboxesStory = {
+  decorators: [withFormik],
+  parameters: {
+    formik: {
+      initialValues: {
+        openForms: {
+          dataSrc: 'variable',
+          itemsExpression: {var: 'someVariable'},
+        },
+      },
+    },
+  },
+};
+
+const sleep = delay => new Promise(resolve => setTimeout(resolve, delay));
+
+export const SelectBoxesResetState: StoryObj<{
+  onSubmit: (values: any) => void;
+}> = {
+  render: ({onSubmit}) => {
+    return (
+      <Formik
+        initialValues={{openForms: {dataSrc: ''}}}
+        onSubmit={(values, helpers) => {
+          onSubmit(values);
+          helpers.setSubmitting(false);
+        }}
+      >
+        {({isSubmitting}) => (
+          <Form noValidate>
+            <ValuesConfig<SelectboxesComponentSchema> name="values" />
+            <button type="submit" className="btn btn-primary">
+              Submit
+            </button>
+            {isSubmitting && 'Submitting...'}
+          </Form>
+        )}
+      </Formik>
+    );
+  },
+  args: {
+    onSubmit: jest.fn(),
+  },
+  play: async ({canvasElement, step, args}) => {
+    const canvas = within(canvasElement);
+
+    const doSubmit = async () => {
+      const btn = canvas.getByRole('button', {name: 'Submit'});
+      // https://github.com/testing-library/user-event/issues/1075
+      // apparently userEvent.click(submitButton) doesn't do anything in Firefox :/
+      fireEvent.click(btn);
+      await waitFor(async () => {
+        await expect(canvas.queryByText('Submitting...')).toBeNull();
+      });
+    };
```

Comment:
I was under the impression that testing was quite nice these days with testing-library, but this issue and realization really made me lose confidence again :/